// @expo/steps/dist_commonjs/BuildFunctionGroup.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createBuildFunctionGroupByIdMapping = exports.BuildFunctionGroup = void 0;
const errors_js_1 = require("./errors.cjs");
class BuildFunctionGroup {
    constructor({ namespace, id, createBuildStepsFromFunctionGroupCall, }) {
        this.namespace = namespace;
        this.id = id;
        this.createBuildStepsFromFunctionGroupCall = createBuildStepsFromFunctionGroupCall;
    }
    getFullId() {
        return this.namespace === undefined ? this.id : `${this.namespace}/${this.id}`;
    }
}
exports.BuildFunctionGroup = BuildFunctionGroup;
function createBuildFunctionGroupByIdMapping(buildFunctionGroups) {
    const buildFunctionGroupById = {};
    for (const buildFunctionGroup of buildFunctionGroups) {
        if (buildFunctionGroupById[buildFunctionGroup.getFullId()] !== undefined) {
            throw new errors_js_1.BuildConfigError(`Build function group with id ${buildFunctionGroup.getFullId()} is already defined.`);
        }
        buildFunctionGroupById[buildFunctionGroup.getFullId()] = buildFunctionGroup;
    }
    return buildFunctionGroupById;
}
exports.createBuildFunctionGroupByIdMapping = createBuildFunctionGroupByIdMapping;
